### PR TITLE
Multiwords vic machine

### PR DESCRIPTION
**Pull request by emlin**
Move vic-machine based on "github.com/urfave/cli", which is simple to use, and support multiple words command.
So vic-machine will follow common command format, to have "--" for long argument, and "-" for short argument.
Also changed "image-store" to full name "image-datastore", "passwd" to "password".

Here is one sample command line: 
```
./bin/vic-machine-linux create --target 10.17.109.50 --user root --password ca\$hc0w --compute-resource /Datacenter/host/VSAN/Resources/line --image-datastore vsanDatastore -name line001 --bridge-network InternalNetwork
 --force 
```
Related to: #897 Fixes: #945

### CODE REVIEW

**Comment by emlin** [comment]
@stuclem this change will impact user document, thanks

File: Makefile [hunk 1]
```diff
@@ -307,15 +307,15 @@ $(bootstrap-staging-debug): isos/bootstrap-staging.sh $(iso-base)
 
 $(vic-machine-linux): $(call godeps,cmd/vic-machine/*.go)
 	@echo building vic-machine linux...
-	@GOARCH=amd64 GOOS=linux $(TIME) $(GO) build $(RACE) --ldflags '-extldflags "-static"' -o ./$@ ./$(dir $<)
+	@GOARCH=amd64 GOOS=linux time -f "t%E" $(GO) build $(RACE) -ldflags "-X github.com/vmware/vic/cmd/vic-machine.BuildID=$(BUILD_NUMBER)" -o ./$@ ./$(dir $<)
```
Comment (caglar10ur) [inline Makefile]: these changes are wrong, guessing you branch is not up2date

**Comment by caglar10ur** [comment]
It will LGTM after you revert your Makefile changes :)

File: Makefile (hunk 1 above)
Comment (emlin) [inline Makefile]: I made wrong code merge, thanks for find it.

**Comment by stuclem** [comment]
@emlin thanks for the heads-up. I am tracking this in https://github.com/vmware/vic/issues/948.

File: cmd/vic-machine/create/create.go [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+	if c.computeResourcePath == "" {
+		return cli.NewExitError("--compute-resource Compute resource path must be specified", 1)
+	}
+
+	if c.imageDatastoreName == "" {
+		return cli.NewExitError("--image-datastore Image datastore name must be specified", 1)
+	}
+
+	if c.cert != "" && c.key == "" {
+		return cli.NewExitError("key cert should be specified at the same time", 1)
```
Comment (andrewtchin) [inline cmd/vic-machine/create/create.go]: "key and cert should be specified at the same time"

**Comment by andrewtchin** [comment]
👍 lgtm

**Comment by dougm** [comment]
@emlin you need to update gvt - with the current version, there would only be *.go files added.

We also agreed a while back that vendor/ changes should be in their own PR.

**Comment by dougm** [comment]
Regarding @caglar10ur 's Makefile comment, I think you just need to add '-static' back to the flags.  Or @caglar10ur , did you have an issue with injecting the build number?

Otherwise, lgtm

**Comment by caglar10ur** [comment]
Oh my makefile comment was for something else. @emlin's first revision was reverting the time cmd related changes back to a previous version by mistake. It's no longer the case in the latest revision

**Comment by emlin** [comment]
rebased

**Pull request merged**